### DISCUSSION TITLE
GPU plugin crashes PC if Runelite is not the first app to use the GPU

### DISCUSSION THREAD

**Discussion by bladerail** [runelite/runelite#18036] [Q&A]
My PC (Windows 10 22H2, Ryzen 5 5600x, Nvidia 3070TI, 32GB DDR4 RAM) has been crashing for the past couple of weeks and I am almost certain that it's a Runelite bug, because it always happens when I'm trying to use Runelite. The crashes do not just shut Runelite down, they crash the PC to black and force a full restart. I have however, managed to get crash-free sessions as well, so I initially thought it was my hardware that was faulty. My computer does not crash when doing any other things such as browsing the web or playing other games that are definitely heavier than Runelite.

I saw a suggestion on this [reddit thread](https://www.reddit.com/r/2007scape/comments/m7h2y3/fyi_full_pc_crashes_stalls_gpu_driver_failure/) to toggle the GPU plugin off and on. When I do this, Runelite does not crash. So, I now do this on every Runelite start up.

Today, I played a game of Age of Empires 4, then after that, wanted to play Runescape. I started up Runelite and tried to toggle the GPU plugin, but before I could get to it, the PC crashed. Bear in mind that I hadn't even logged in, I was literally trying to to go the plugin settings to try and toggle it, and it crashed before I could do so.

I have not played anything else the past few days, where I did not experience crashes. But when I think back to the crashes from a couple of weeks back, they seem to always happen AFTER playing a different game, or using the GPU elsewhere, such as Da Vinci Resolve for movie editing. 

I have done the following:
- Re-installed latest Nvidia Graphics Driver (version 556.12) without uninstalling the old one.
- Re-installed latest Nvidia Graphics (version 556.12) after using DDC to remove the existing driver.
- Ran Memtest86 for 8 passes over 8 hours to check if my RAM was faulty. It was fine.
- Ran Heaven's Benchmark to see if my GPU was seemingly failing under load. It was fine.
- I am able to provide Event Viewer error/critical logs if required, but my review tells me that they are not very useful.
  - I do have a couple of Error and 1 Critical statement after each crash, but none of them seem to be helpful. There's always one about a bugcheck, a Critical one about Kernel-Power.
  - I do have a new one worth investigating recently about nvlddmkm but it suggests a hardware problem and I currently do not think that my hardware has any issues because of the tests I ran above.
- I am able to provide Runelite logs or launcher logs if required, but there are no ERROR messages at any point, even the log files for when the PC crashes. They are always INFO level logs.

Any suggestions would be appreciated

**Comment by ericgeoff** [comment]
Any update on this? Similar experience here

**Reply by GunnerStone** [reply]
Experiencing similar issues here only with Runelite seemingly hard crashing my computer. 

Runelite.exe is crashing inside nvoglv64.dll - Nvidia's OpenGL driver. Exception 0xc0000409 (stack buffer overrun in the driver). 

This triggers a TDR which is what all the errors in event viewer are for. Not necessarily faulty hardware, just nvidia trying to revive our dead driver & failing.

**Reply by Maxie35** [reply]
Similar experience here, almost daily now my GPU drivers completely crash causing my PC to enter a 'slideshow', everything moving at like 1 frame per 3 seconds. Restarting GPU drivers ( win + ctrl + shift + B) fixes it. Tried setting V-Sync off in GPU plugin, tried almost every setting in gpu plugin and Nvidia App but I still keep getting this.
